File: client/src/actions/patients.js
import axios from 'axios';

export const postPatient = (patient) => {
    return async (dispatch) => {
        const body = {
            name: patient.name,
            last_name: patient.lastName,
            age: patient.age,
            provider: patient.provider,
            email: patient.email
        }
        try {
            const res = await axios.post(`/patients`, body);
            patient['id'] = res.data.id;
            dispatch({type: 'POST_PATIENT', payload: patient})
        } catch (err) {
            console.log((err.response && err.response.data)) 
        }
    }
} 

export const getPatients = () => {
    return async (dispatch) => {
        try {
            const res = await axios.get(`/patients`);
            console.log(res)
            const patients = res.data
            for (const patient of patients) {
                patient.lastName = patient.last_name;
            }
            dispatch({type: 'GET_PATIENTS', payload: res.data});
        } catch (err) {
            console.log((err.response && err.response.data));
        }
    }
}

File: api/models/patients.py
import sqlite3
from datetime import datetime

from models import sql_commands

def post_patient(patient):
    patient['created_date'] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    row = ( 
        patient['name'], 
        patient['last_name'], 
        patient['age'], 
        patient['provider'], 
        patient['email'],
    )
    query = 'INSERT INTO patients (name, last_name, age, provider, email) VALUES (?, ?, ?, ?, ?)'

    patient_id = sql_commands.sql_execute_post(query, row)

    return patient_id


def get_patient(patient_id, fields=[]):
    if fields:
        fields_str = ', '.join(fields)
    else:
        fields_str = '*'

    query = "SELECT {} FROM patients WHERE id = {}".format(fields_str, patient_id) 

    query_result = sql_commands.sql_execute_get_list(query)

    if not query_result:
        return None

    patient = query_result[0] # is a list with only one element; a dict with the requiered user 
    return patient


def get_patients():
    query = 'SELECT * FROM patients'

    query_result = sql_commands.sql_execute_get_list(query)

    return query_result
